### DISCUSSION TITLE
Improve E2 Timers

### DISCUSSION THREAD

**Discussion by Denneisk** [wiremod/wire#2672] [Suggestions]
Discussion about making new timers that are more sensible and fit with modern E2. Mostly just writing it here so it can be documented.

The most obvious syntax is probably something like `timer(number name, number periodInSeconds, function callback)` so you could do
```ts
function myFunc() {
    print("function executed")
}

if(first()) {
    timer("myTimer", 3, myFunc)
}
```

**Comment by Fasteroid** [comment]
Would be even better with lambdas or a function datatype.  Imagine if you could do this:
```golo
FuncPtr = function what(){
    print("how")
}
FuncPtr()

timer( "myTimer", FuncPtr, 3 )

Wtf = array()
Wtf[1, function] = FuncPtr
```

**Reply by Denneisk** [reply]
Clairvoyant
